demo/src/task-scheduler.ts

```diff
@@ -0,0 +1,40 @@
+// Faster setTimeout(fn, 0) implementation using postMessage API
+// Based on https://dbaron.org/log/20100309-faster-timeouts
+export class MicroTaskScheduler {
+  readonly messageName = 'zero-timeout-message';
+
+  private executionQueue: Function[];
```

Comment:
You can initialize the variable while declaring it, e.g. 
`private executionQueue: Array<...> = []`, so you don't have to repeat the type in the constructor (that's a nice syntactic sugar offered by typescript)